src/main/java/org/ice4j/stack/MessageProcessor.java

```diff
@@ -38,60 +37,52 @@
     private static final Logger logger
         = Logger.getLogger(MessageProcessor.class.getName());
 
-    /**
-     * The listener that will be collecting error notifications.
-     */
-    private final ErrorHandler errorHandler;
-
-    /**
-     * The queue where we store incoming messages until they are collected.
-     */
-    private final BlockingQueue<RawMessage> messageQueue;
-
-    /**
-     * The listener that will be retrieving <tt>MessageEvent</tt>s
-     */
-    private final MessageEventHandler messageEventHandler;
-
     /**
      * The <tt>NetAccessManager</tt> which has created this instance and which
      * is its owner.
      */
     private final NetAccessManager netAccessManager;
 
     /**
-     * The flag that indicates whether we are still running.
+     * Ram message which is being processed
      */
-    private boolean running = false;
+    private final RawMessage rawMessage;
 
     /**
-     * A reference to the thread that we use to execute ourselves.
+     * The listener that will be collecting error notifications.
      */
-    private Thread runningThread = null;
+    private final ErrorHandler errorHandler;
+
+    /**
+     * The listener that will be retrieving <tt>MessageEvent</tt>s
+     */
+    private final MessageEventHandler messageEventHandler;
 
     /**
      * Creates a Message processor.
      *
      * @param netAccessManager the <tt>NetAccessManager</tt> which is creating
      * the new instance, is going to be its owner, specifies the
-     * <tt>BlockingQueue</tt> which is to store incoming messages, specifies the
      * <tt>MessageEventHandler</tt> and represents the <tt>ErrorHandler</tt> to
      * handle exceptions in the new instance
+     * @param message the <tt>RawMessage</tt> to be asynchronously processed by
+     * this MessageProcessor
      * @throws IllegalArgumentException if any of the mentioned properties of
      * <tt>netAccessManager</tt> are <tt>null</tt>
      */
-    MessageProcessor(NetAccessManager netAccessManager)
+    MessageProcessor(
```

Comment:
@bbaldino yes, the meaning of class is slightly changed, earlier it was started thread which reads from queue. Now this class represent kind of "job" which is put into queue along with message which need to be processed. I used `ExecutorService` to done queued processing, but then in need `Runnable` to put into `ExecutorService` queue, so I've reused `MessageProcessor` as it was already `Runnable`.

Reply:
Gotcha, will this be done for every packet though?  Have you noticed if creating one of these per packet has affected performance at all?

Reply:
> Gotcha, will this be done for every packet though? 

Yes for now in this pull request `MessageProcessor` is created per `RawMessage`. Essentially `MessageProcessor` is just a `Runnable` wrapping `RawMessage` to make it possible to put `RawMessage` processing into `ExecutorService`s queue. 

>  Have you noticed if creating one of these per packet has affected performance at all?

I did not notice performance issue due to increased allocations count for `MessageProcessor`.
What I've noticed is a performance gain due to decreased number of threads, see tables below, which compares versions without this pull request and with this pull request for `20` audio only conferences with `3` peers each.

### baseline 
| Name | Value | Comment
|-|-|-|
|Computer:                          | 3 GHz Intel Core i7 mac mini, osx 10.14	| 
|JVB version:                       | 1.1-20180615.220358-69 |
|ice4j version:                     | 2.0.0-20181105.145933-16 | # with PR 154 (not merged yet)
|org.ice4j.TERMINATION_DELAY, ms    | 604800000 |
|Conf per server:                   | 20 |
|Peers per conf:                    | 3 |
|Est. RTP streams:                  | 180 |
|CPU load:                          | 58-68 % |
|Active threads:                    | 974 |
|Network IN:                        | ~ 2538 KBit/s |
|Network OUT:                       | ~ 4892 KBit/s |
|Average RTT:                       | ~3-20 ms |
|ICE connection stability:          | stable	|

### baseline + PR 155
| Name | Value | Comment
|-|-|-|
|Computer:                          | 3 GHz Intel Core i7 mac mini, osx 10.14	| 
|JVB version:                       | 1.1-20180615.220358-69 |
|ice4j version:                     | 2.0.0-20181105.145933-16 | # with PR 154 (not merged yet), 155 (this PR)
|org.ice4j.TERMINATION_DELAY, ms    | 604800000 |
|Conf per server:                   | 20 |
|Peers per conf:                    | 3 |
|Est. RTP streams:                  | 180 |
|CPU load:                          | 50-64% | # notice ~`4%` drop here
|Active threads:                    | 789 |
|Network IN:                        | ~ 2431 KBit/s |
|Network OUT:                       | ~ 4878 KBit/s |
|Average RTT:                       | ~3-28 ms |
|ICE connection stability:          | stable	|


Actually such short-living allocations like `MessageProcessor` are cheap both when allocated and cheap for GC collection, since `MessageProcessor` object is short living and will not likely to survive first generation which is cheap to collect.

Since you worried about allocation I'll add a pool of `MessageProcessor` objects per `NetAccessManager`, so allocations will go down, but probably `MessageProcessor` objects will live in older generations eventually. This might have an impact, but I'm not sure how hard to measure it.

Reply:
> Since you worried about allocation I'll add a pool of MessageProcessor objects per NetAccessManager, so allocations will go down, but probably MessageProcessor objects will live in older generations eventually. This might have an impact, but I'm not sure how hard to measure it.

@bbaldino  I've added pooling of `MessageProcessor` [into separate branch on my fork](https://github.com/mstyura/ice4j/pull/1/files) to estimate complexity how it could be implemented.
If you think it worth using pool despite slightly increased complexity - I'll update current PR to include changes.
Thank you very much for review and comments!

Reply:
In case I'm lucky enough today 😸 such that pull request is acceptable, but pooling is decided to be included and I'm not responding due to my timezone, [feel free to merge](https://github.com/mstyura/ice4j/pull/1/) branch in my repository, so this pull request will also be updated to include pooling.

Reply:
After all the changes I like version with pooling more, so adding it by default in this PR.